pkg/apic/apiservice.go

```diff
@@ -31,6 +32,7 @@ func buildAPIServiceSpec(serviceBody *ServiceBody) mv1a.ApiServiceSpec {
 }
 
 func (c *ServiceClient) buildAPIService(serviceBody *ServiceBody) *mv1a.APIService {
+	ownerObject, ownerErr := c.getOwnerObject(serviceBody, true)
```

Comment:
Maybe instead of calling getOwnerObject for every resource we can add the object and err to the serviceBody? Then just use those attributes directly?